src/lib.rs

```diff
@@ -831,6 +831,8 @@ impl<L, R> DerefMut for Either<L, R>
 impl<L, R> Error for Either<L, R>
     where L: Error, R: Error
 {
+    #[allow(deprecated)]
+    #[allow(bare_trait_objects)]
```

Comment:
I guess you copied these from `fn cause`, but the `bare_trait_objects` lint doesn't apply here.

Reply:
@cuviper , I removed it.